.github/workflows/flow-macos.yml

```diff
@@ -0,0 +1,130 @@
+name: CI Full suite - MACOS
```

Comment:
1. change name 
2. use at the same mechanism to upload macos-12 artifacts like we did for JSON